ToneMapper/Model/Infastracture/TMToneAdjustFilter.fsh

```diff
@@ -3,9 +3,16 @@
 
 varying lowp vec2 textureCoordinates;
 uniform sampler2D texture;
+uniform sampler2D smoothTexture;
+uniform sampler2D smootherTexture;
+uniform lowp float mediumContrast;
+uniform lowp float fineContrast;
 uniform lowp mat4 adjustmentMatrix;
-uniform lowp float contrast;
 
 void main() {
-  gl_FragColor = adjustmentMatrix * texture2D(texture, textureCoordinates);
+  lowp vec4 preGlobalAdjustmentColor =
+      (1.0 + mediumContrast + fineContrast) * texture2D(texture, textureCoordinates) -
+      fineContrast * texture2D(smoothTexture, textureCoordinates) -
```

Comment:
It's quite confusing i'll admit, but actually
`I_details = I_orig - I_very-smooth` and `I_fine-details = I_orig - I_smooth` and in efforts of getting 
```
I_orig + a * I_details + b * I_fine-details 
```
I do
```
(1 + a + b) * I_orig - b * I_very-smooth - a * I_smooth
```